market-contract/tests/functions/core/withdraw_protocol_fee.rs

```diff
@@ -0,0 +1,164 @@
+        let id1 = contract
+            .with_account(&user1.wallet)
+            .await?
+            .open_order(base_amount, /*AssetType::Base,*/ OrderType::Buy, price)
+            .await?
+            .value;
+
+        let accumulated_protocol_fee = contract.total_protocol_fee().await.unwrap().value;
+        assert_eq!(accumulated_protocol_fee, 0);
+
+        let cancel_order_tx0 = contract
+            .with_account(&user0.wallet)
+            .await?
+            .cancel_order(id0)
+            .await;
+        let cancel_order_tx1 = contract
+            .with_account(&user1.wallet)
+            .await?
+            .cancel_order(id1)
+            .await;
+        assert!(cancel_order_tx0.is_ok());
+        assert!(cancel_order_tx1.is_ok());
+
+        let accumulated_protocol_fee = contract.total_protocol_fee().await.unwrap().value;
+        assert_eq!(accumulated_protocol_fee, 0);
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn assert_protocol_fee_accumulates_after_match_order() -> anyhow::Result<()> {
+        let defaults = Defaults::default();
+        let (contract, user0, user1, assets) = setup(
+            defaults.base_decimals,
+            defaults.quote_decimals,
+            defaults.price_decimals,
+        )
+        .await?;
+
+        let to_quote_scale =
+            10_u64.pow(defaults.price_decimals + defaults.base_decimals - defaults.quote_decimals);
+        let price = 70_000_000_000_000_u64; // 70,000$ price
+        let base_amount = 100_000_u64; // 0.001 BTC
+        let quote_amount = price / to_quote_scale * base_amount;
+        contract
+            .with_account(&user0.wallet)
+            .await?
+            .deposit(base_amount, assets.base.id)
+            .await?;
+        contract
+            .with_account(&user1.wallet)
+            .await?
+            .deposit(quote_amount, assets.quote.id)
+            .await?;
+
+        let accumulated_protocol_fee = contract.total_protocol_fee().await.unwrap().value;
+        assert_eq!(accumulated_protocol_fee, 0);
+
+        let id0 = contract
+            .with_account(&user0.wallet)
+            .await?
+            .open_order(
+                base_amount,
+                /*AssetType::Base,*/ OrderType::Sell,
+                price,
+            )
+            .await?
+            .value;
+        let id1 = contract
+            .with_account(&user1.wallet)
+            .await?
+            .open_order(base_amount, /*AssetType::Base,*/ OrderType::Buy, price)
+            .await?
+            .value;
+
+        let accumulated_protocol_fee = contract.total_protocol_fee().await.unwrap().value;
+        assert_eq!(accumulated_protocol_fee, 0);
+
+        contract.match_order_pair(id0, id1).await?;
+
+        let accumulated_protocol_fee = contract.total_protocol_fee().await.unwrap().value;
+        assert!(accumulated_protocol_fee != 0);
+
+        let withdraw_tx = contract.withdraw_protocol_fee(user0.identity()).await;
```

Comment:
check that fuel asset was actually withdrawn. check user and contract balance. and total_protocol_fee should be 0 after

Reply:
Fixed